### ISSUE TITLE
Facebook Graph API Update

### ISSUE THREAD

**Issue by lpcruz** [Sntax/facebookEventAPI#1]
Hey there, 

I'm interested in utilizing this script — do you know if it's up-to-date with Facebook's Graph API updates?

RE: Latest

https://developers.facebook.com/docs/graph-api/using-graph-api/v2.5

Let me know - Thanks!

**Comment by andrewjrhill** [comment]
I haven't touched this code since I wrote it 5 months ago. It should be functional still unless facebook has made some big changes.

**Comment by lpcruz** [comment]
@Sntax Thanks, Andrew. I'll give it a go and keep you posted if I run into any issues. 

Best, 

-L

**Comment by andrewjrhill** [comment]
Sure thing. Keep me posted. Interested to see how you use it! Good luck!

**Comment by lpcruz** [comment]
Hey @Sntax - One more thing for clarification. 

I noticed that the following code seems to only apply to Facebook Fan Pages as opposed to Facebook profiles:

```
// Find unique page ID by viewing source of Facebook page and searching for page_id.
// Example: View page source on https://www.facebook.com/GitHub/, CTRL+F to find
// page_id will return the numeric page ID '262588213843476' for GitHubs' Facebook Page.
var fbPageId = '827714403940306';
```

In this specific example it looks like `var fbPageID` refers to the following instance: 

https://www.facebook.com/eracapetown/

If I were to view the page source of the actual event page or a user profile, I'm unable to parse out `page_id` Any thoughts?

Basically, is there a way to substitute that variable with the exact ID of the event page?

**Comment by andrewjrhill** [comment]
@lpcruz Not sure I understand your question correctly. The code used in this repo is indeed for eracapetown so if I view their facebook events page directly (view-source:https://www.facebook.com/eracapetown/events) I can still do a CTRL-F and find a page_id in that code (in this instance 827714403940306). Perhaps send me a URL of a facebook page you're trying to grab the ID from?

**Comment by lpcruz** [comment]
@Sntax Sure thing - sorry if that was unclear. See:

https://www.facebook.com/events/1648479002077996/

If I'm understanding this correctly, is the url that actually gets parsed follow this schema:

`https://www.facebook.com/{{ user-id }}/events`

**Comment by lpcruz** [comment]
Ah sorry - I missed the following:

```
  // Constructed URL used to retrieve JSON objects containing our events.
    var eventsURL = 'https://graph.facebook.com/' + fbPageId + '/events/' +
                    '?fields=' + fields +
                    '&access_token=' + accessToken;
```

**Comment by lpcruz** [comment]
Got it to work. Out of curiosity, does `accessToken` need to be periodically refreshed?

**Comment by andrewjrhill** [comment]
@lpcruz Glad to hear it! I haven't needed to refresh the token yet. This might shed some light on limitations: https://developers.facebook.com/docs/sharing/opengraph/using-actions. Good Luck!
